lib/cucumber/step_match.rb

```diff
@@ -139,4 +139,17 @@ def activate(test_step)
       return test_step
     end
   end
+
+  class AmbiguousStepMatch
+
+    def initialize(error)
+      @error = error
+    end
+
+    def activate(test_step)
+      return test_step.with_action { raise Core::Test::Result::Ambiguous.new(@error.message) }
```

Comment:
Given that return value of a Ruby method is the last thing evaluated? No, it's just there because that's how someone wrote the code that I am basing this code on.

I don't necessarily understand why established patterns are there, but I can at least follow them as I attempt to fix code that I am entirely unfamiliar with. :)